### ISSUE TITLE
Command /bin/sh failed with exit code 1 - After upgrading reactnative to 0.15.0 and NPM to 3.3.6

### ISSUE THREAD

**Issue by vinu123** [facebook/react-native#4493]
I can no longer compile in Xcode after upgrading React Native to 0.15.0 and NPM to 3.3.6. When I compile Xcode throws an error saying "Command /bin/sh failed with exit code 1". 

Forgive me if this has been covered elsewhere.

**Comment by gre** [comment]
Having the same issue.. I really wish this can be fixed. also there is lot of dups of this issue ( https://github.com/facebook/react-native/issues/3948 )

**Comment by gorangajic** [comment]
I have similar or the same or similar issue, after trying to create new project 

```sh
++ export MANPATH
++ export PATH
++ hash -r
++ export NVM_PATH=/Users/null/.nvm/versions/node/v4.2.1/lib/node
++ NVM_PATH=/Users/null/.nvm/versions/node/v4.2.1/lib/node
++ export NVM_BIN=/Users/null/.nvm/versions/node/v4.2.1/bin
++ NVM_BIN=/Users/null/.nvm/versions/node/v4.2.1/bin
++ '[' '' = true ']'
++ nvm_is_iojs_version v4.2.1
+++ echo v4.2.1
+++ cut -c1-5
++ '[' _v4.2. = _iojs- ']'
++ echo 'Now using node v4.2.1'
+ react-native bundle --entry-file index.ios.js --platform ios --dev true --bundle-output /Users/null/Library/Developer/Xcode/DerivedData/PredictClient-gninzjqnjlmrdgemcxrwgqfknfnl/Build/Products/Debug-iphonesimulator/PredictClient.app/main.jsbundle --assets-dest /Users/null/Library/Developer/Xcode/DerivedData/PredictClient-gninzjqnjlmrdgemcxrwgqfknfnl/Build/Products/Debug-iphonesimulator/PredictClient.app
bundle: Created ReactPackager
uncaught error Error: SyntaxError: /Users/null/dev/TestApp/node_modules/react-native/Libraries/ReactNative/ReactNative.js: Line 74: Unsupported type annotation type: FunctionTypeAnnotation
  72 | };
  73 | 
> 74 | var render = function(
     |              ^
  75 |   element: ReactElement,
  76 |   mountInto: number,
  77 |   callback?: ?(() => void)
    at File.errorWithNode (/Users/null/dev/TestApp/node_modules/react-native/node_modules/babel-core/lib/transformation/file/index.js:489:13)
    at NodePath.errorWithNode (/Users/null/dev/TestApp/node_modules/react-native/node_modules/babel-core/lib/traversal/path/index.js:155:26)
    at NodePath.Function (/Users/null/dev/TestApp/babel-plugin-typecheck/lib/index.js:48:22)
    at NodePath.call (/Users/null/dev/TestApp/node_modules/react-native/node_modules/babel-core/lib/traversal/path/context.js:56:28)
    at NodePath.visit (/Users/null/dev/TestApp/node_modules/react-n
Command /bin/sh failed with exit code 1
```

**Comment by satya164** [comment]
It seems to be fixed in 0.16-rc - https://github.com/facebook/react-native/commit/9a64f17de84e50a4fc60f1b96168aa61925934a8
